chain/jsonrpc/src/lib.rs

```diff
@@ -342,8 +356,18 @@ impl JsonRpcHandler {
                 let query_response = self.query(rpc_query_request).await;
                 process_query_response(query_response)
             }
-            "status" => self.status().await,
-            "tx" => self.tx_status_common(request.params, false).await,
+            "status" => {
+                let status_response = self.status().await?;
+                serde_json::to_value(status_response)
+                    .map_err(|err| RpcError::parse_error(err.to_string()))
```

Comment:
Hm, why is this pares error rather than serialization error / Internal error? From the look of it, seems like serialization should return internal error or maybe just `.unwrap` even, and rely on the framework to catch panics and 500? Assuming that serialization failure indicates a bug in our code.